store/store_helper.go

```diff
@@ -0,0 +1,57 @@
+package store
+
+import (
+	types "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+// GetAndDecode gets and decodes key and returns it. Returns nil if key doesn't exist.
+func GetAndDecode[T any](store types.KVStore, dec func([]byte) (T, error), key []byte) (T, error) {
+	var res T
+	bz := store.Get(key)
+	if bz == nil {
+		return res, nil
+	}
+	resp, err := dec(bz)
+	if err != nil {
+		return resp, err
+	}
+	return resp, err
+}
+
+// GetAndDecodeWithBool gets and decodes key and returns it. Returns false if key doesn't exist.
+func GetAndDecodeWithBool[T any](store types.KVStore, dec func([]byte) (T, bool), key []byte) (T, bool) {
```

Comment:
what do you think about combining both `GetAndDecode` and `GetAndDecodeWithBool`? Or what's the reasoning behind having 2 separate functions for this?

Reply:
and if we need two, maybe GetAndDecodeWithBool could be a higher-order function of GetAndDecode? What do you think @likhita-809 ?

Reply:
> what do you think about combining both `GetAndDecode` and `GetAndDecodeWithBool`? Or what's the reasoning behind having 2 separate functions for this?

we can't use the same func `GetAndDecode` which returns `error` at places where we are required to return a boolean value. If used, it returns weird errors/ correct value isn't returned

Reply:
@facundomedica or @JeancarloBarrios please resolve the response makes sense :)